ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -1469,14 +1498,15 @@ private List<SerializableXAResourceDeserializer> getXAResourceDeserializers() {
         }
         return serializableXAResourceDeserializers;
     }
-	
+
 	protected XAResource _theXAResource;
 
 	private RecoverableXAConnection _recoveryObject;
 	private Xid _tranID;
 	private boolean _prepared;
 	private boolean _committed;
 	private boolean _valid;
+	private boolean _rolledBack;
```

Comment:
I am not sure if it's important or not but I do not think you are making this part of the persistent state of the object so after recovery I suppose an extra rollback could be called?

Reply:
@jmfinelli please you add a comment here summarising if you feel it's important (or not) to (at least try to, as it won't always be possible if various crash takes place) persist the information of whether the rollback took place or whether you know it would not be useful in a crash recovery situation?

Reply:
Replied ^^^